osquery/tables/system/uptime.cpp

```diff
@@ -0,0 +1,69 @@
+/*
+ *  Copyright (c) 2014, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#include <osquery/tables.h>
+
+#if defined(__APPLE__)
+  #include <time.h>
+  #include <errno.h>
+  #include <sys/sysctl.h>
+#elif defined(__linux) || defined(__linux__) || defined(linux)
```

Comment:
We do have an undocumented set of defines that allow us to specify the platform from CMake logic. One day we should codemod out all the compiled-based decision logic (e.g.: `gcc -dM -E - < /dev/null`) and prefer our CMake decisions as the source of truth.